velox/exec/PartitionedOutput.h

```diff
@@ -47,17 +46,17 @@ class Destination {
     rows_.push_back(rows);
   }
 
-  BlockingReason advance(
+  virtual BlockingReason advance(
       uint64_t maxBytes,
       const std::vector<vector_size_t>& sizes,
       const RowVectorPtr& output,
       PartitionedOutputBufferManager& bufferManager,
       bool* FOLLY_NONNULL atEnd,
-      ContinueFuture* FOLLY_NONNULL future);
+      ContinueFuture* FOLLY_NONNULL future) = 0;
 
-  BlockingReason flush(
+  virtual BlockingReason flush(
       PartitionedOutputBufferManager& bufferManager,
```

Comment:
Do we want to include PartitionedOutputBufferManager in the interface? It looks like this an implementation details of a Presto-specific Destination.

Reply:
Good point. This was the original way flush was written but you are right. I can make flush to be independent or not have flush in the interface.